### PR TITLE
Add docs about running pdb in houston app

### PR DESCRIPTION
**Pull request by karenc**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -332,3 +332,41 @@ For example if you have set up your admin user using a different email address,
...
+
+or
+
+```python
+import pdb; pdb.set_trace()
+```
+
+See the debugger commands here: https://docs.python.org/3/library/pdb.html#debugger-commands
+
+If you want to debug the running houston service, you need to change `tasks/codex/run.py`:
```
Comment (hwindsor) [inline CONTRIBUTING.md:352]: Worth adding that this does not work for code running in celery tasks

**Review by hwindsor — APPROVED** [review]
Add the bit about celery tasks and this is good to go

File: CONTRIBUTING.md (hunk 1 above)
Comment (karenc) [reply CONTRIBUTING.md:352]: Ah right I'll add it

Comment (karenc) [reply CONTRIBUTING.md:352]:
```diff
diff --git a/CONTRIBUTING.md b/CONTRIBUTING.md
index 7abc2427..f9a5e010 100644            
--- a/CONTRIBUTING.md                      
+++ b/CONTRIBUTING.md                      
@@ -368,5 +368,9 @@ Stop the running houston service and start it in foreground:
                                           

 docker-compose stop houston               
-docker-compose run -p 83:5000 --rm houston
+docker-compose run --rm -p 83:5000 --name=houston houston
                      
+                                          
+The debugger doesn't work for celery tasks but you can try to run the task in
+the foreground by removing the `.delay()` part.  Or add logs in the celery task
+which should show in `docker-compose logs -f celery_worker`.
```

Comment (karenc) [reply CONTRIBUTING.md:352]: @hwindsor Removing `.delay()` from `sage_detection` in `AssetGroupSighting` causes some errors but I'm unable to figure out why :shrug: so I guess we can't run `sage_detection` in foreground.

```
2021-11-11 16:38:03,927 [INFO] [app.extensions.restManager.RestManager] Sending post request to ACM: http://acm:5000/api/engine/detect/cnn/lightnet
2021-11-11 16:38:03,927 [DEBUG] [app.extensions.restManager.RestManager] Contents {'params': {'endpoint': '/api/engine/detect/cnn/lightnet/', 'jobid': '1e38bc79-ed6a-484f-a2e1-cc868119a2c6', 'callback_url': 'houston+http://houston:5000/api/v1/asset_groups/sighting/a3e67cb1-8e9a-43db-a057-ca050e5aadae/sage_detected/1e38bc79-ed6a-484f-a2e1-cc868119a2c6', 'callback_detailed': True, 'input': {'start_detect': '/api/engine/detect/cnn/lightnet/', 'labeler_algo': 'densenet', 'labeler_model_tag': 'zebra_v1', 'model_tag': 'ggr2', 'nms_thresh': 0.4, 'sensitivity': 0.4}, 'image_uuid_list': ['houston+http://houston:5000/api/v1/assets/src_raw/57ce62a1-0fdb-4a98-8ba3-24d63dab2d61']}, 'headers': {'Accept': '*/*', 'Content-Type': 'application/json', 'User-Agent': 'python-requests/2.26.0'}}
2021-11-11 16:38:04,040 [INFO] [werkzeug] 172.31.0.1 - - [11/Nov/2021 16:38:04] "POST /api/v1/asset_groups/ HTTP/1.1" 500 -
Traceback (most recent call last):
  File "/code/app/modules/asset_groups/resources.py", line 110, in post
    asset_group.begin_ia_pipeline(metadata)
  File "/code/app/modules/asset_groups/models.py", line 1448, in begin_ia_pipeline
    self.ensure_repository()
  File "/code/app/modules/asset_groups/models.py", line 1493, in ensure_repository
    repo = self.get_repository()
  File "/code/app/modules/asset_groups/models.py", line 1488, in get_repository
    repo_path = pathlib.Path(self.get_absolute_path())
  File "/code/app/modules/asset_groups/models.py", line 1355, in get_absolute_path
    asset_group_path = os.path.join(asset_group_database_path, str(self.guid))
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/orm/attributes.py", line 294, in __get__
    return self.impl.get(instance_state(instance), dict_)
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/orm/attributes.py", line 725, in get
    value = state._load_expired(state, passive)
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/orm/state.py", line 652, in _load_expired
    self.manager.deferred_scalar_loader(self, toload)
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/orm/loading.py", line 942, in load_scalar_attributes
    raise orm_exc.DetachedInstanceError(
sqlalchemy.orm.exc.DetachedInstanceError: Instance <AssetGroup at 0x7f50aa47ceb0> is not bound to a Session; attribute refresh operation cannot proceed (Background on this error at: http://sqlalche.me/e/13/bhk3)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/code/flask_restx_patched/namespace.py", line 505, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/webargs/core.py", line 459, in wrapper
    return func(*new_args, **kwargs)
  File "/code/flask_restx_patched/namespace.py", line 252, in dump_wrapper
    response = func(*args, **kwargs)
  File "/code/app/modules/asset_groups/resources.py", line 119, in post
    asset_group.delete()
  File "/code/app/modules/asset_groups/models.py", line 1467, in delete
    for asset in self.assets:
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/orm/attributes.py", line 294, in __get__
    return self.impl.get(instance_state(instance), dict_)
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/orm/attributes.py", line 730, in get
    value = self.callable_(state, passive)
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/orm/strategies.py", line 717, in _load_for_state
    raise orm_exc.DetachedInstanceError(
sqlalchemy.orm.exc.DetachedInstanceError: Parent instance <AssetGroup at 0x7f50aa47ceb0> is not bound to a Session; lazy load operation of attribute 'assets' cannot proceed (Background on this error at: http://sqlalche.me/e/13/bhk3)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 2464, in __call__
    return self.wsgi_app(environ, start_response)
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 2450, in wsgi_app
    response = self.handle_exception(e)
  File "/usr/local/lib/python3.9/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.9/site-packages/flask_restx/api.py", line 671, in error_router
    return original_handler(f)
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1867, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.9/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/usr/local/lib/python3.9/site-packages/flask_restx/api.py", line 669, in error_router
    return self.handle_error(e)
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/usr/local/lib/python3.9/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/usr/local/lib/python3.9/site-packages/flask_restx/api.py", line 671, in error_router
    return original_handler(f)
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/local/lib/python3.9/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/usr/local/lib/python3.9/site-packages/flask_restx/api.py", line 669, in error_router
    return self.handle_error(e)
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/usr/local/lib/python3.9/site-packages/flask_restx/api.py", line 403, in wrapper
    resp = resource(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/flask/views.py", line 89, in view
    return self.dispatch_request(*args, **kwargs)
  File "/usr/local/lib/python3.9/site-packages/flask_restx/resource.py", line 49, in dispatch_request
    resp = meth(*args, **kwargs)
  File "/code/flask_restx_patched/namespace.py", line 515, in wrapper
    AuditLog.houston_fault(None, str(ex))
  File "/code/app/extensions/logging/__init__.py", line 140, in houston_fault
    cls.audit_log(logger, msg, cls.AuditType.HoustonFault, *args, **kwargs)
  File "/code/app/extensions/logging/__init__.py", line 85, in audit_log
    cls._log_message(logger, msg, *args, **kwargs)
  File "/code/app/extensions/logging/__init__.py", line 67, in _log_message
    msg = f'{msg} executed by user :{current_user.guid} {current_user.email}'
  File "/usr/local/lib/python3.9/site-packages/werkzeug/local.py", line 348, in __getattr__
    return getattr(self._get_current_object(), name)
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/orm/attributes.py", line 294, in __get__
    return self.impl.get(instance_state(instance), dict_)
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/orm/attributes.py", line 725, in get
    value = state._load_expired(state, passive)
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/orm/state.py", line 652, in _load_expired
    self.manager.deferred_scalar_loader(self, toload)
  File "/usr/local/lib/python3.9/site-packages/sqlalchemy/orm/loading.py", line 942, in load_scalar_attributes
    raise orm_exc.DetachedInstanceError(
sqlalchemy.orm.exc.DetachedInstanceError: Instance <User at 0x7f50aa504700> is not bound to a Session; attribute refresh operation cannot proceed (Background on this error at: http://sqlalche.me/e/13/bhk3)
```

Comment (hwindsor) [reply CONTRIBUTING.md:352]: Agreed. There is something weird going on and I think it needs investigating but importantly it is not part of this PR and this PR should go in as it.

**Review by colinwkingen — APPROVED** [review]
LGTM

**Comment by colinwkingen** [comment]
Pushing through the tests to save time as no code is altered in the PR

**Pull request merged**